flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/SubpartitionTestBase.java

```diff
@@ -182,7 +182,7 @@ private static void assertNextBufferOrEvent(
 			ResultSubpartitionView readView,
 			int expectedReadableBufferSize,
 			boolean expectedIsBuffer,
-			Class<? extends AbstractEvent> expectedEventClass,
+			@Nullable Class<? extends AbstractEvent> expectedEventClass,
```

Comment:
I would prefer for this method to return `Buffer` and:
```
void assertNextBuffer(...) {
  Buffer buffer = assertNextBufferOrEvent(...);
  assertTrue(buffer.isBuffer());
  buffer.recycleBuffer();
}

void assertNextEvent(...) {
  Buffer buffer = assertNextBufferOrEvent(...);
  assertFalse(buffer.isBuffer());
  assertThat(EventSerializer.fromBuffer(buffer, ...), ...);
  buffer.recycleBuffer();
}
```

btw, you are not recycling the buffer in case of failure

Reply:
I added the recycling in case of failures but kept the `@Nullable` because otherwise this would need some more duplication with this cleanup